packages/app/src/app/pages/Dashboard/Content/routes/Settings/TeamSettings/WorkspaceSettings.tsx

```diff
@@ -76,11 +82,50 @@ export const WorkspaceSettings = () => {
   const [inviteValue, setInviteValue] = useState('');
   const [inviteLoading, setInviteLoading] = useState(false);
 
+  const [newMemberAuthorization, setNewMemberAuthorization] = React.useState<
+    TeamMemberAuthorization
+  >(
+    // get TeamMemberAuthorization key by value
+    Object.keys(TeamMemberAuthorization).find(key => {
+      const value = TeamMemberAuthorization[key];
+      return value === team?.settings.defaultAuthorization;
+    }) as TeamMemberAuthorization
+  );
+
+  const numberOfEditors = team.userAuthorizations.filter(
+    member => member.authorization !== TeamMemberAuthorization.Read
+  ).length;
+
+  // A workspace can have unused seats in their subscription
+  // if they have already paid for X editors for the YEARLY plan
+  // then removed some members from the team
+  const numberOfUnusedSeats =
+    team.subscription &&
+    team.subscription.billingInterval === SubscriptionBillingInterval.Yearly
+      ? team.subscription.quantity - numberOfEditors
+      : 0;
+
+  // if the user is going to be charged for adding a member
+  // throw them a confirmation modal
+  const confirmNewMemberAddition =
+    team?.subscription &&
+    team?.subscription.origin !== WorkspaceSubscriptionOrigin.Pilot &&
+    numberOfUnusedSeats === 0 &&
+    team.settings.defaultAuthorization !== TeamMemberAuthorization.Read;
+  const confirmMemberRoleChange =
+    team?.subscription &&
+    team?.subscription.origin !== WorkspaceSubscriptionOrigin.Pilot &&
+    numberOfUnusedSeats === 0;
+
```

Comment:
Maybe this should be derived state?

Here are some examples: https://github.com/codesandbox/codesandbox-client/blob/fix-closing.issue/packages/app/src/app/overmind/namespaces/dashboard/state.ts#L141

Let me know if you want help here

Reply:
Some of this is dependent on local state, so keeping it in this one place. If/when we need to use it on another screen, it would be wise to move to global state